internal/sourcemap/metadata.go

```diff
@@ -110,34 +110,24 @@ func maybeParseURLPath(s string) string {
 	return url.Path
 }
 
-func (s *MetadataCachingFetcher) update(ms []Metadata) {
+func (s *MetadataCachingFetcher) update(ms map[Identifier]Metadata) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	for _, m := range ms {
-		if contentHash, ok := s.set[m.id]; ok {
-			// seen
-			delete(s.set, m.id)
-
+	for id, contentHash := range s.set {
+		if metadata, ok := ms[id]; ok {
 			// content hash changed, invalidate the sourcemap cache
-			if contentHash != m.contentHash {
-				s.invalidationChan <- m.id
+			if contentHash != metadata.contentHash {
+				s.invalidationChan <- id
 			}
-		}
-	}
-
-	// Loop for any unseed sourcemap
-	for k := range s.set {
-		// the sourcemap no longer exists in ES.
-		// invalidate the sourcemap cache.
-		s.invalidationChan <- k
+		} else {
+			// the sourcemap no longer exists in ES.
+			// invalidate the sourcemap cache.
+			s.invalidationChan <- id
 
-		// remove from metadata cache
-		delete(s.set, k)
-	}
-
-	for _, m := range ms {
-		s.set[m.id] = m.contentHash
+			// remove from metadata cache
+			delete(s.set, id)
+		}
```

Comment:
I am not following where the values are actually updated in the set now.

It might need to be something like this:
```
	for id, contentHash := range s.set {
		if metadata, ok := ms[id]; ok {
			// content hash changed, invalidate the sourcemap cache
			if contentHash != metadata.contentHash {
				s.invalidationChan <- id
				// remove from metadata cache
				delete(s.set, id)

			} else {
				// remove entry from map if already part of the cache set and the content is unchanged
				delete(ms, id)
			}
		} else {
			// the sourcemap no longer exists in ES.
			// invalidate the sourcemap cache.
			s.invalidationChan <- id

			// remove from metadata cache
			delete(s.set, id)
		}
	}
	for id, m := range ms {
		s.set[id] = m.contentHash
	}
```

Reply:
My mistake, Thanks for spotting this! I've pushed some changes!